grype/db/v5/matcher/java/maven_search.go

```diff
@@ -14,13 +17,24 @@ import (
 // MavenSearcher is the interface that wraps the GetMavenPackageBySha method.
 type MavenSearcher interface {
 	// GetMavenPackageBySha provides an interface for building a package from maven data based on a sha1 digest
-	GetMavenPackageBySha(string) (*pkg.Package, error)
+	GetMavenPackageBySha(context.Context, string) (*pkg.Package, error)
 }
 
 // mavenSearch implements the MavenSearcher interface
 type mavenSearch struct {
-	client  *http.Client
-	baseURL string
+	client      *http.Client
+	baseURL     string
+	rateLimiter *rate.Limiter
+}
+
+// newMavenSearch creates a new mavenSearch instance with rate limiting
+// rate is specified as 1 request per 300ms
+func newMavenSearch(client *http.Client, baseURL string) *mavenSearch {
+	return &mavenSearch{
+		client:      client,
+		baseURL:     baseURL,
+		rateLimiter: rate.NewLimiter(rate.Every(300*time.Millisecond), 1), // 1 request per 300ms, using the integration test to determine the best rate.
```

Comment:
just out of curiosity, why 1 request per 300 ms? from experimentation? or did you find any useful maven docs on this?

Reply:
yeah through experimenting, I found that it needed a few iterations sometimes to get the 403 but using 300ms I've not seen one failure.  There could be other tuning, I tried a bunch of http client settings too but this was the lowest I found that reliably worked.